src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -105,6 +104,7 @@ export class Match {
 
 		return {
 			before,
+			fullBefore,
 			inside,
 			after,
 		};
```

Comment:
@andreamah `before` is truncated at the beginning for the search results view so the match has a chance of being visible. The search results view also truncates the end automatically with CSS ellipsis. The hover title is expected to provide stuff that has been cut. That's why we need `fullBefore`. I'm sure it could be named better, though, your guidance will be appreciated.

The size of `fullBefore` may want to be constrained, but here it becomes interesting, or in the words of a relevant comment in existing code, fishy:

* The `before`, `inside` and `after` elements are constrained in `searchModel.ts` to 250 chars (`MAX_PREVIEW_CHARS`). Perhaps it would make sense to constrain `fullBefore` to the same 250, cutting from the start, of course.
* The code in `searchResultsView.ts` constrains the hover title to 999 characters. 
* Search results seem to be constrained in length already in the constructor of `TextSearchMatch` (`src/vs/workbench/services/search/common/search.ts`, around line 311 where it performs "elision"). That code sees itself as "fishy", but seems to remove the need for us to shorten `fullBefore` ourselves.

There seems to be an inordinate amount of shortening going on; bug #198566 complains that it's being overdone. I'm trying to make the search results more useful when the match is beyond 26 characters into a normal-length line.

What do you think, do we need to shorten `fullBefore`?

Reply:
That sounds fine. I think that we can rely on the elision to trim the before enough. Thanks for looking into this!